src/main/java/teammates/ui/webapi/action/GetSessionResultsAction.java

```diff
@@ -37,14 +38,18 @@ public void checkSpecificAccessControl() {
         Intent intent = Intent.valueOf(getNonNullRequestParamValue(Const.ParamsNames.INTENT));
         switch (intent) {
         case INSTRUCTOR_RESULT:
-            InstructorAttributes instructor = logic.getInstructorForGoogleId(courseId, userInfo.getId());
+            if (userInfo == null || !userInfo.isInstructor) {
+                throw new UnauthorizedAccessException("Instructor account is required to access this resource");
+            }
+            InstructorAttributes instructor = logic.getInstructorForGoogleId(courseId, userInfo.id);
             gateKeeper.verifyAccessible(instructor, fs);
             break;
         case STUDENT_RESULT:
+            if (userInfo == null && StringHelper.isEmpty(getRequestParamValue(Const.ParamsNames.REGKEY))) {
```

Comment:
Removing this null check throws a NPE in `verifyInaccessibleWithoutLogin` under `testAccessControl_withoutCorrectAuthInfoAccessStudentResult_shouldFail`

Reply:
Then it's time to figure out which part of code result in the NPE. (Does the NPE cause by accessing `userInfo` without checking login privilege? If yes, should the method be responsible of checking login privilege?)

Reply:
The NPE was caused by calling the action's `checkSpecificAccessControl` from the test case `verifyInaccessibleWithoutLogin` which specifically logs the user out. Since the action here is generated from the `ActionFactory` in `WebApiServlet`, it's feasible that someone could attempt to manually submit a request to our backend to request for the resource mapped to this action. Currently the server manages this by throwing an NPE since it can't get the `userInfo`.

Reply:
Ahh. I see. I thought it is the same code related to feedback submission while it is not.

Reply:
Yeap, the feedback submission has no issues when I remove the extra checks as you pointed out. I'll rectify that first and request for your review afterwards.